<!-- bootcamp-header: generated by tools/build_headers.py, edit the README timetable instead -->
# Solutions: Python and variables

Worked solutions to the exercises in [`02_EX_Python_and_Variables.ipynb`](https://github.com/mikekestemont/dtaantwerp26-27.github.io/blob/DTA_Bootcamp_2026_students/exercises/questions/02_EX_Python_and_Variables.ipynb), which go with the session [`02_W1_Tue_Python_and_Variables.ipynb`](https://github.com/mikekestemont/dtaantwerp26-27.github.io/blob/DTA_Bootcamp_2026_students/notebooks/02_W1_Tue_Python_and_Variables.ipynb).  

[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/mikekestemont/dtaantwerp26-27.github.io/blob/DTA_Bootcamp_2026_students/exercises/solutions/02_SOL_Python_and_Variables.ipynb)

placeholder

Every exercise is followed by one possible solution (sometimes two). Other correct solutions exist: what matters is that yours works and that you understand why.

## Core exercises

### 1. Your name (basic)

Create a variable called `first_name` and give it your own first name as a value. Then print the variable with the `print()` function.

In [1]:
first_name = 'Emma'
print(first_name)

Emma


### 2. Your full name (basic)

Create another variable, `last_name`, with your surname as its value. Concatenate the two strings into a new variable `full_name` and print it. Make sure there is a space between the first name and the surname.

In [2]:
last_name = 'Woodhouse'
full_name = first_name + ' ' + last_name
print(full_name)

Emma Woodhouse


### 3. Pages in a book (basic)

A book has 24 chapters of 18 pages each. Store both numbers in variables with clear names, multiply them and store the result in a third variable. Print that variable to check whether the multiplication worked.

In [3]:
chapters = 24
pages_per_chapter = 18
pages = chapters * pages_per_chapter
print(pages)

432


### 4. Multiplying a string (basic)

The `*` operator also works on strings. Use it to print a line of 40 dashes (`-`), and then to print the word `'echo'` five times.

In [4]:
print('-' * 40)
print('echo' * 5)

----------------------------------------
echoechoechoechoecho


Multiplying a string by a number repeats it. Adding two strings glues them together (concatenation); we will see in the next session that the other arithmetic operators do not work on strings.

### 5. Make it True (standard)

Change the cell so that the output is `True`. There are two ways: change the value of `x`, or change the value of `y`. Can you explain why the cell says `False` now?

In [5]:
x = 12
y = 3 * 4
x == y

True

In [6]:
x = '12'
y = '12'
x == y

True

`x` was the *string* `'12'` (note the quotation marks), whereas `y` was the *number* 12. A string is never equal to a number, even if it looks the same when printed.

### 6. Valid and invalid names (standard)

The cell below does not run. Explain, for each line, whether the variable name is allowed and why (not), and then fix the lines that are wrong so that the cell runs.

In [7]:
# The original cell, with an explanation per line. This cell produces an error on purpose.
my-name = 'Tommy'      # hyphens are not allowed in names, only underscores: my_name
river_fish_ = 'herring' # fine (an underscore at the end is allowed)
300 = 'number of Spartans'  # a name cannot start with a digit (and a number cannot be a name)
Equals10 = 15          # fine (digits are allowed after the first character)
"age" = 46             # a string cannot be a name: remove the quotation marks
Time = 'Noon"          # not a name problem: the quotation marks around the string do not match
Cruise = Hardy = Hanks = 'Tom'  # fine: three names bound to the same value at once

SyntaxError: unterminated string literal (detected at line 7) (3361644981.py, line 7)

In [8]:
# The corrected cell
my_name = 'Tommy'
river_fish_ = 'herring'
spartans = 300
Equals10 = 15
age = 46
Time = 'Noon'
Cruise = Hardy = Hanks = 'Tom'
print(my_name, spartans, Time, Hanks)

Tommy 300 Noon Tom


### 7. Tickets (standard)

A poetry reading has 94 visitors. 46 of them are students, who pay 8.50 euros; everybody else pays 10.50 euros. Calculate how much all tickets together cost and print the result. Use the variables that are given.

In [9]:
visitors = 94
students = 46
student_price = 8.5
full_price = 10.5

total = students * student_price + (visitors - students) * full_price
print(total)

895.0


### 8. A film title (standard)

The first cell below runs, the second one does not. Explain why, and change the second cell so that it prints `12 monkeys`.

In [10]:
brother = 'Hansel '
sister = 'Gretel'
siblings = brother + sister
siblings

'Hansel Gretel'

In [11]:
# This cell produces an error on purpose: you cannot add a number and a string
amount = 12
animals = ' monkeys'
film_title = amount + animals

TypeError: unsupported operand type(s) for +: 'int' and 'str'

In [12]:
amount = 12
animals = ' monkeys'
film_title = str(amount) + animals   # str() turns the number into the string '12'
film_title

'12 monkeys'

`+` means addition for numbers and concatenation for strings, and Python refuses to guess which one you mean when you mix the two. Converting the number to a string with `str()` is called *casting*; the next session says more about it.

## Extra exercises

### 9. Wholesale price (standard)

The cover price of a book is 24.95 euros, but bookshops get a 40 percent discount. Shipping costs 3 euros for the first copy and 75 cents for each additional copy. What is the total cost of an order of 60 copies? Print the result in a readable sentence, using `str()` where necessary.

In [13]:
cover_price = 24.95
discount = 0.40
copies = 60

price_per_copy = cover_price * (1 - discount)
shipping = 3 + 0.75 * (copies - 1)
total = copies * price_per_copy + shipping

print('The order costs ' + str(total) + ' euros.')

The order costs 945.4499999999999 euros.


The result has a long tail of decimals (`945.4499999999999`): computers store decimal numbers with a tiny imprecision. This is normal, and later we will see how to round numbers when printing them.

### 10. Spot the errors (standard)

Each line in the cell below is meant to print `A message`. Which lines contain an error, and what is wrong? Correct them.

In [14]:
print("A message")   # the full stop after the closing bracket was wrong
print("A message")   # the quotation marks did not match (a double one to open, single ones to close)
print('A message"')  # nothing wrong here: a double quotation mark *inside* single ones is just a character

A message
A message
A message"


### 11. How long is a novel? (standard)

A novel has 320 pages; a page has on average 32 lines, and a line about 11 words. Estimate the number of words in the novel in one line of code, store the result in a variable with a clear name, and print it. Then estimate how many hours it takes to read the novel at 250 words per minute.

In [15]:
pages = 320
lines_per_page = 32
words_per_line = 11

number_of_words = pages * lines_per_page * words_per_line
print(number_of_words)

reading_hours = number_of_words / 250 / 60
print(reading_hours)

112640
7.509333333333333


### 12. Swap (standard)

Write code that swaps the values of the two variables below, so that `first_number` becomes 5 and `second_number` becomes 3. Hint: you need an extra variable.

In [16]:
first_number = 3
second_number = 5

temporary = first_number
first_number = second_number
second_number = temporary

print(first_number, second_number)

5 3


Without the extra variable, the first assignment would overwrite one of the values before you have copied it. (Python has a shortcut for this, `first_number, second_number = second_number, first_number`, which you will meet later.)

### 13. Predict the output (harder)

Think about what the three `print()` statements will show, *before* you run the cell. Then run it and check.

In [17]:
x = 3
y = 15

b = y        # b is 15
y = x        # y is 3
x = b - 1    # x is 14

print(b)
print(y)
print(x)

15
3
14


### 14. Brackets (harder)

Add brackets to the expression so that `x` becomes 0 and the last line prints `True`. Do not change the numbers or the operators.

In [18]:
x = 15 - 6 * 2 - (1 + 2)

x == 0

True

### 15. Leftover books (harder)

Ten books are divided evenly among three students. How many books are left over? Look up online what the `%` operator ("modulo") does and use it. Bonus: solve the same problem *without* `%`, using `//` (integer division) instead, and check your solution with a different number of books.

In [19]:
books = 10
students = 3

leftover = books % students
print(leftover)

1


In [20]:
# Bonus: without the modulo operator
leftover = books - (books // students) * students
print(leftover)

1


`%` gives the remainder of a division; `//` gives the result of the division rounded down to a whole number. Both come back in the next session.